src/Http/Routing/src/Builder/EndpointRouteBuilderExtensions.cs

```diff
@@ -495,8 +498,28 @@ public static RouteHandlerBuilder MapFallback(this IEndpointRouteBuilder endpoin
         // explicit about the MethodInfo representing the "handler" and not the RequestDelegate?
 
         // Add MethodInfo as metadata to assist with OpenAPI generation for the endpoint.
+        // This is added directly to the builder to ensure it's first in the metadata collection.
         builder.Metadata.Add(handler.Method);
 
+        // Defer adding attributes on the handler to the call to RDF as those automatically generated by the
+        // RDF have a higher specificity.
+        var attributes = handler.Method.GetCustomAttributes();
+
+        // This can be null if the delegate is a dynamic method or compiled from an expression tree
+        if (attributes is not null)
+        {
+            foreach (var attribute in attributes)
+            {
+                metadataToAdd.Add(attribute);
```

Comment:
What do you think about adding the `MethodInfo`'s attributes at the start of `RequestDelegateFactory.Create()` instead of here? I think it makes sense to have RDF do it now that it's adding a bunch of other metadata. That wasn't the case originally.